kimchi/src/circuits/argument.rs

```diff
@@ -0,0 +1,23 @@
+//! An argument is simply a number of constraints,
+//! which we want to enforce on all points of the domain.
+//! Both the permutation and the plookup arguments fit this type.
+//! Gates can be seen as filtered arguments,
+//! which apply only in some points (rows) of the domain.
+
+use ark_ff::FftField;
+
+use crate::{
+    alphas::Alphas,
+    circuits::expr::{ConstantExpr, Expr},
+};
+
+/// The interface for a minimal argument implementation
+pub trait Argument {
```

Comment:
Perhaps it's not the intention here, but a `Gate` interface could also expose the selector, so we could combine that automatically